### PR TITLE
[project-base] Add graphic checkboxes and radiobuttons

### PR DESCRIPTION
**Pull request by TomasGottvald**
| Q             | A
| ------------- | ---
|Description, reason for the PR| Add graphic version of checkboxes and radiobuttons.
|New feature| No 
|[BC breaks](https://github.com/shopsys/shopsys/blob/master/docs/contributing/backward-compatibility-promise.md)| No 
|Fixes issues| ... 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

Screen before change:
obrazek

Screen after change:
obrazek

### CODE REVIEW

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Login/loginForm.html.twig [hunk 1]
```diff
@@ -38,17 +38,15 @@
             <dt></dt>
             <dd>
                 <label class="form-choice">
-                    <span class="form-choice__input">
-                        {{ form_widget(form.rememberMe) }}
-                    </span>
-                    <span class="form-choice__label">
+                    {{ form_widget(form.rememberMe, { attr: { class: "css-checkbox" } }) }}
+                    <label class="css-checkbox__image form-choice__label" for="{{ form.rememberMe.vars.id }}">
```
Comment (topolanekmartin) [inline project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Login/loginForm.html.twig]: Label inside label? Is it necessary?

File: project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+input[type=checkbox].css-radio, input[type=radio].css-radio {
+    position: absolute;
+    height: 1px;
+    width: 1px;
+    left: -1000px;
+    margin: -1px;
+    padding: 0;
+    z-index: -1000;
+
+    overflow: hidden;
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less]: Yes, that is true. Moved to top block

File: project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+            width: @custom-inputs-icon-size;
+            height: @custom-inputs-icon-size;
+
+            background-repeat: no-repeat;
+            background-position: left top;
+            background-size: 3*@custom-inputs-icon-size 2*@custom-inputs-icon-size;
+        }
+
+        &:hover {
+
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less]: Removed

File: project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+            &:before {
+                background-position-x: center;
+            }
+        }
+    }
+
+    &:focus, &:active {
+
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less]: Removed

File: project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+        & + .css-checkbox__image, & + .css-radio__image {
+
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less]: Removed

File: project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+            &:before {
+                background-position-x: center;
+            }
+        }
+    }
+
+    &:checked {
+
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less]: Removed

File: project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+        & + .css-checkbox__image, & + .css-radio__image {
+
```
[the remark this answers is not in the document]
Comment (TomasGottvald) [reply project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less]: Removed

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Login/loginForm.html.twig (hunk 1 above)
Comment (TomasGottvald) [reply project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Login/loginForm.html.twig]: In our usage we do not need label in label. So I changed form-line to div element.

File: project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less (hunk 2 above)
[the remark this answers is not in the document]
Comment (topolanekmartin) [reply project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less]: Fine. Thanks.

File: project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less (hunk 3 above)
[the remark this answers is not in the document]
Comment (topolanekmartin) [reply project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less]: Thanks.

File: project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less (hunk 4 above)
[the remark this answers is not in the document]
Comment (topolanekmartin) [reply project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less]: Thanks.

File: project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less (hunk 5 above)
[the remark this answers is not in the document]
Comment (topolanekmartin) [reply project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less]: Thanks.

File: project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less (hunk 6 above)
[the remark this answers is not in the document]
Comment (topolanekmartin) [reply project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less]: Thanks.

File: project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less (hunk 7 above)
[the remark this answers is not in the document]
Comment (topolanekmartin) [reply project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less]: Thanks.

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Login/loginForm.html.twig (hunk 1 above)
Comment (topolanekmartin) [reply project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Login/loginForm.html.twig]: That's good. Thanks.

**Comment by henzigo** [comment]
Hello @pesektomas. Maybe you can found inspiration about graphic checkboxes in tests [here](https://github.com/shopsys/shopsys/pull/1245)

**Comment by pesektomas** [comment]
> Hello @pesektomas. Maybe you can found inspiration about graphic checkboxes in tests [here](https://github.com/shopsys/shopsys/pull/1245)

Thank you @henzigo. I see the difference that I choose the checkbox by javascript and in the link are choose by css. But. Solution in PR is nicer that my. I will change it.

**Comment by TomasGottvald** [comment]
I have just added docs for graphic checkboxes

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Order/step3.html.twig [hunk 8]
```diff
@@ -8,132 +8,127 @@
     <div class="web__line">
         <div class="web__container">
 
-            <div class="box-order">
-                <div class="box-order__info">
-                    <h2>{{ 'Personal data'|trans }}</h2>
+    <div class="box-order">
```
[the remark this answers is not in the document]
Comment (TomasLudvik) [reply project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Order/step3.html.twig]: This is not solved

[the remark this answers is not in the document]
Comment (TomasGottvald) [reply project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Order/step3.html.twig]: Fixed. thx

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Inline/Newsletter/subscription.html.twig [hunk 9]
```diff
@@ -15,20 +15,16 @@
...
             {{ form_errors(form, { errors_attr: { class : 'in-message--alert' } }) }}
             <div class="form-choice">
-                <div class="form-choice__input">
-                    {{ form_widget(form.privacyPolicyAgreement) }}
-                </div>
-                <div class="form-choice__label">
-                    <label for="{{ form.privacyPolicyAgreement.vars.id }}">
+                {{ form_widget(form.privacyPolicyAgreement, { attr: { class: "css-checkbox" } }) }}
+                <label class="css-checkbox__image form-choice__label" for="{{ form.privacyPolicyAgreement.vars.id }}">
                         {% if privacyPolicyArticle is not null %}
```
Comment (pesektomas) [inline project-base/src/Shopsys/ShopBundle/Resources/views/Front/Inline/Newsletter/subscription.html.twig]: One more bad indentation

Comment (TomasGottvald) [reply project-base/src/Shopsys/ShopBundle/Resources/views/Front/Inline/Newsletter/subscription.html.twig]: thx for info, fixed

**Comment by TomasGottvald** [comment]
> 
> 
> Upgrade notes are missing

I have just added upgrade notes. Please check it. Thx

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 10]
```diff
@@ -542,6 +542,69 @@ There you can find links to upgrade notes for other versions too.
...
+        ```
+
+    - `project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/components/in/icon.less` (line 13)
+        ```diff
+          font-size: 16px;
+        + height: 16px;
+          cursor: help;
+        ```
+
+    - add new file `project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less` according to pull request
```
Comment (pesektomas) [inline upgrade/UPGRADE-v8.1.0-dev.md]: Could you add link to file here please? `https://github.com/shopsys/shopsys/blob/8.1/project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/core/form/custom-inputs.less`

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 11]
```diff
@@ -542,6 +542,69 @@ There you can find links to upgrade notes for other versions too.
...
+    - link this new file in `project-base/src/Shopsys/ShopBundle/Resources/styles/front/common/main.less`
```
Comment (pesektomas) [inline upgrade/UPGRADE-v8.1.0-dev.md]: Could you add diff here, pleas?

```
  @import "core/form/input.less";
+ @import "core/form/custom-inputs.less";
  @import "core/form/btn.less";
```

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 12]
```diff
@@ -542,6 +542,69 @@ There you can find links to upgrade notes for other versions too.
...
+        +            {{ form_label(form, label, { label_attr: { class: "css-c+heckbox__image" }}) }}
+        +            {% set errors_attr = errors_attr|default({})|merge({'class': (errors_attr.class|default('form-error--choice'))}) %}
+        +            {{ form_errors(form, { errors_attr: errors_attr } ) }}
+        +            {{ block('icon') }}
+        +        </div>
+        +     </div>
+          {% endblock checkbox_row %}
+     ```
+
+    - update all twig files according to pull request or find all checkboxes and add them `{ attr: { class: "css-checkbox" } }` and direct after input tag add label with `class="css-checkbox__image"`. You can change class name to `css-radio` and `css-radio__image` according to input type.
```
Comment (pesektomas) [inline upgrade/UPGRADE-v8.1.0-dev.md]: Could you put list of twig files?

File: upgrade/UPGRADE-v8.1.0-dev.md [hunk 13]
```diff
@@ -542,6 +542,69 @@ There you can find links to upgrade notes for other versions too.
...
+    - you can use tag `<span class="css-checkbox_image">` instead of label, if label wraps form input and text
+    - update all test according to pull request - so it can accept new graphic inputs
```
Comment (pesektomas) [inline upgrade/UPGRADE-v8.1.0-dev.md]: Could you put list of tests here? List is:

- [OrderCest.php](https://github.com/shopsys/shopsys/blob/8.1/project-base/tests/ShopBundle/Acceptance/acceptance/OrderCest.php)
- [OrderPage.php](https://github.com/shopsys/shopsys/blob/8.1/project-base/tests/ShopBundle/Acceptance/acceptance/PageObject/Front/OrderPage.php)
- [ProductFilterPage.php](https://github.com/shopsys/shopsys/blob/8.1/project-base/tests/ShopBundle/Acceptance/acceptance/PageObject/Front/ProductFilterPage.php)
- [RegistrationPage.php](https://github.com/shopsys/shopsys/blob/8.1/project-base/tests/ShopBundle/Acceptance/acceptance/PageObject/Front/RegistrationPage.php)
- [ProductFilterCest.php](https://github.com/shopsys/shopsys/blob/8.1/project-base/tests/ShopBundle/Acceptance/acceptance/ProductFilterCest.php)

File: upgrade/UPGRADE-v8.1.0-dev.md (hunk 10 above)
Comment (TomasGottvald) [reply upgrade/UPGRADE-v8.1.0-dev.md]: added

File: upgrade/UPGRADE-v8.1.0-dev.md (hunk 11 above)
Comment (TomasGottvald) [reply upgrade/UPGRADE-v8.1.0-dev.md]: added

File: upgrade/UPGRADE-v8.1.0-dev.md (hunk 12 above)
Comment (TomasGottvald) [reply upgrade/UPGRADE-v8.1.0-dev.md]: added

File: upgrade/UPGRADE-v8.1.0-dev.md (hunk 13 above)
Comment (TomasGottvald) [reply upgrade/UPGRADE-v8.1.0-dev.md]: added

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Registration/register.html.twig [hunk 14]
```diff
@@ -24,15 +24,12 @@
...
-                            <div class="form-choice__input">
-                                {{ form_widget(form.privacyPolicy) }}
-                            </div>
-                            <div class="form-choice__label">
-                                <label for="{{ form.privacyPolicy.vars.id }}">
+                    <dl class="form-line">
+                        <dt></dt>
+                        <dd>
+                            <div class="form-choice">
+                                {{ form_widget(form.privacyPolicy, { attr: { class: "css-checkbox" } }) }}
```
Comment (s3tezsky) [inline project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Registration/register.html.twig]: Would it be possible to put checkbox under label tag?

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Product/filterFormMacro.html.twig [hunk 15]
```diff
@@ -117,11 +113,9 @@
                                     {{ parameterForm.vars.label }}:
                                 </span>
                                 {% for parameterValueForm in parameterForm %}
-                                    <label class="form-choice js-product-filter-parameter-value">
-                                        <span class="form-choice__input">
-                                            {{ form_widget(parameterValueForm) }}
-                                        </span>
-                                        <span class="form-choice__label">
+                                    <div class="form-choice js-product-filter-parameter-value">
+                                        {{ form_widget(parameterValueForm, { attr: { class: 'css-checkbox'} }) }}
```
Comment (s3tezsky) [inline project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Product/filterFormMacro.html.twig]: Would be possible to put checkbox under label tag?

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Login/windowForm.html.twig [hunk 16]
```diff
@@ -12,18 +12,16 @@
         </div>
 
         <div class="box-login__item box-login__item--checkbox">
-            <label class="form-choice">
-                <span class="form-choice__input">
-                    {{ form_widget(form.rememberMe) }}
-                </span>
-                <span class="form-choice__label">
+            <div class="form-choice">
+                {{ form_widget(form.rememberMe, { attr: { class: "css-checkbox" } }) }}
```
Comment (s3tezsky) [inline project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Login/windowForm.html.twig]: Would be possible to put checkbox under label tag?

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Login/loginForm.html.twig [hunk 17]
```diff
@@ -36,19 +36,17 @@
                 <dl class="form-line">
                     <dt></dt>
                     <dd>
-                        <label class="form-choice">
-                            <span class="form-choice__input">
-                                {{ form_widget(form.rememberMe) }}
-                            </span>
-                            <span class="form-choice__label">
+                        <div class="form-choice">
+                            {{ form_widget(form.rememberMe, { attr: { class: "css-checkbox" } }) }}
```
Comment (s3tezsky) [inline project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Login/loginForm.html.twig]: Would be possible to put checkbox under label tag?

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Order/step3.html.twig [hunk 18]
```diff
@@ -77,39 +77,34 @@
...
-                                    </label>
-                                    <span class="js-validation-errors-list js-validation-error-list-{{ form.legalConditionsAgreement.vars.id }} form-error form-error--choice display-none">
-                                        <span class="form-error__icon">
-                                            <i class="svg svg-warning"></i>
-                                        </span>
-                                        <ul class="form-error__list"></ul>
+                    <div class="form-line">
+                        <div class="form-choice">
+                            <div class="form-choice__label">
+                                {{ form_widget(form.legalConditionsAgreement, { attr: { class: "css-checkbox" } }) }}
```
Comment (s3tezsky) [inline project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Order/step3.html.twig]: Would be possible to put checkbox under label tag?

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Product/filterFormMacro.html.twig [hunk 19]
```diff
@@ -25,11 +25,9 @@
         </div>
 
         <div class="box-filter__choice">
-            <label class="form-choice">
-                <span class="form-choice__input">
-                    {{ form_widget(filterForm.inStock) }}
-                </span>
-                <span class="form-choice__label">
+            <div class="form-choice">
+                {{ form_widget(filterForm.inStock, { attr: { class: "css-checkbox" } }) }}
```
Comment (s3tezsky) [inline project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Product/filterFormMacro.html.twig]: Would be possible to put checkbox under label tag?

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Product/filterFormMacro.html.twig [hunk 20]
```diff
@@ -51,11 +49,9 @@
                     {{ 'Flags'|trans }}:
                 </span>
                 {% for flagForm in filterForm.flags %}
-                    <label class="form-choice">
-                        <span class="form-choice__input">
-                            {{ form_widget(flagForm) }}
-                        </span>
-                        <span class="form-choice__label">
+                    <div class="form-choice">
+                        {{ form_widget(flagForm, { attr: { class: "css-checkbox" } }) }}
```
Comment (s3tezsky) [inline project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Product/filterFormMacro.html.twig]: Would be possible to put checkbox under label tag?

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Product/filterFormMacro.html.twig [hunk 21]
```diff
@@ -81,9 +77,9 @@
                     {{ 'Brand'|trans }}:
                 </span>
                 {% for brandForm in filterForm.brands %}
-                    <label class="form-choice">
-                        {{ form_widget(brandForm, { attr: { class: 'form-choice__input'} }) }}
-                        <span class="form-choice__label">
+                    <div class="form-choice">
+                        {{ form_widget(brandForm, { attr: { class: 'css-checkbox'} }) }}
```
Comment (s3tezsky) [inline project-base/src/Shopsys/ShopBundle/Resources/views/Front/Content/Product/filterFormMacro.html.twig]: Would be possible to put checkbox under label tag?

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Form/theme.html.twig [hunk 22]
```diff
@@ -49,22 +49,15 @@
...
-                    {% set errors_attr = errors_attr|default({})|merge({'class': (errors_attr.class|default('form-error--choice'))}) %}
-                    {{ form_errors(form, { errors_attr: errors_attr } ) }}
-                    {{ block('icon') }}
-                </div>
-            </div>
-        </dd>
-    </dl>
+    <div class="{{ rowClass|default('form-line') }}">
+        <div class="form-choice">
+            {{ form_widget(form, { attr: { class: "css-checkbox" } }) }}
```
Comment (s3tezsky) [inline project-base/src/Shopsys/ShopBundle/Resources/views/Front/Form/theme.html.twig]: Would be possible to put checkbox under label tag?

File: project-base/src/Shopsys/ShopBundle/Resources/views/Front/Inline/Newsletter/subscription.html.twig [hunk 23]
```diff
@@ -15,20 +15,16 @@
...
-                <div class="form-choice__label">
-                    <label for="{{ form.privacyPolicyAgreement.vars.id }}">
-                        {% if privacyPolicyArticle is not null %}
-                            {% set articleUrl = url('front_article_detail', {'id': privacyPolicyArticle.id}) %}
-                            {{ "I agree with <a href=\"%url%\" target=\"_blank\">privacy policy</a>."|trans({'%url%': articleUrl})|raw }}
-                        {% else %}
-                            {{ 'I agree with privacy policy.'|trans }}
-                        {% endif %}
-                    </label>
+                {{ form_widget(form.privacyPolicyAgreement, { attr: { class: "css-checkbox" } }) }}
```
Comment (s3tezsky) [inline project-base/src/Shopsys/ShopBundle/Resources/views/Front/Inline/Newsletter/subscription.html.twig]: Would be possible to put checkbox under label tag?

**Comment by TomasGottvald** [comment]
> 
> 
> Hi @TomasGottvald, can you please check transport and payments?
> 
> Snímek obrazovky 2019-11-18 v 11 36 38

Fixed checkbox position. Please check it. Thx

**Pull request merged**